src/unitxt/templates.py

```diff
@@ -14,12 +13,21 @@ class Template(StreamInstanceOperator):
     """The role of template is to take the fields of every instance and verbalize it.
 
     Meaning the template is taking the instance and generating source, target and references.
+
+    Args:
+        skip_rendered_instance (bool): if "source", "target", and "references" are already defined fields in the instance, skip its processing
+        postprocessors: a list of strings being artifact names of text processors, to be applied on the model output
+        instruction: a formatting string that yields an instruction with potential participation of values from the "inputs" part of the instance
```

Comment:
There is a delicate issue here.  Since instruction will not be rendered per sample (e.g. in few shot), it can not use  any fields that are not the same of in all instances.   It can use fields which are the same over all instances. 

For example :

Classify the sentiment of this text to one of the following classes: positive, negative.  <-- classes aer common

Text: do do
Sentiment: positive

Text: bla bla
Sentiment: negative

Text: bla bla
Sentiment: negative

Text: foo foo
Sentiment:

Reply:
So we should validate, when formatting, that all instructions from the sample and demo instances are the same.

Reply:
StandardRecipe invokes Operator Template before Operator AddDemosField (see lines 150-160 of [standard.py](https://github.com/IBM/unitxt/pull/565/files#diff-a4122a708bc5f51b81dc0316fb8034bff822104b0e9d82bbe30d44761979ec36) . So the demo instances have the very same instructions as the instances of train, test, and validation. 
Does this answer your concern?

Reply:
Now looking at this, I wanted to ask you: is this OK that 
```
if self.augmentor.augment_model_input:
            self.steps.append(self.augmentor)
```
occurs after the demos are planted in train and validation and test? the augmentors will not affect the demos that already sit inside each instance.  Is this alright?